### DISCUSSION TITLE
How to turn off auto @swc/helpers alias

### DISCUSSION THREAD

**Discussion by jayjliang** [web-infra-dev/rsbuild#3711] [Q&A] [answered]
rsbuild会自动注入alias配置，将@swc/helpers指向@rsbuild/core所有目录的node_modules，如果项目中有其他包依赖了不同版本的@swc/helpers，例如0.3.x，构建就会报错，因为0.5.x的版本和0.3.x的版本有break-change

**Comment by chenjiahan** [comment, marked as answer]
Try this:


```js
// rsbuild.config.ts
export default defineConfig({
  tools: {
    rspack(config) {
      if (config.resolve?.alias) {
        delete config.resolve.alias['@swc/helpers'];
      }
    }
  }
});
```

**Reply by jayjliang** [reply]
貌似不行
Module not found: Can't resolve '@swc/helpers/_/_object_spread_props'
自动插入的polyfill会变成这样的代码，但是这样的引用是不存在的

**Reply by chenjiahan** [reply]
You can disable the `externalHelpers` option of SWC (this may increase the bundle size since the SWC helpers will be injected multiple times)

```js
export default defineConfig({
  tools: {
    swc: {
      jsc: {
        externalHelpers: false,
      },
    },
  },
});
```

https://swc.rs/docs/configuration/compilation#jscexternalhelpers

**Comment by chenjiahan** [comment]
In the next release (https://github.com/web-infra-dev/rsbuild/pull/3715):

```js
export default {
  source: {
    alias: (alias) => {
      delete alias['@swc/helpers'];
    },
  },
};
```
